### PR TITLE
Adding divide two numbers

### PR DESCRIPTION
**Pull request by Djokwa**
---
name: solution review 
about: A template PR for code review with a checklist
---

### Files

- [ ] The file name describes the function's behavior
- [ ] There is a module docstring in the function file
- [ ] The test file's name matches the function file name -
  `/tests/test_file_name.py`
- [ ] There is a module docstring in the tests file

### Unit Tests

- [ ] The test class has a helpful name in PascalCase
- [ ] The test class has a docstring
- Every unit test has
  - [ ] A helpful name
  - [ ] A clear docstring
  - [ ] Only one assertion
  - [ ] There is no logic in the unit test
- [ ] All tests pass
- [ ] There are tests for defensive assertions
- [ ] There are tests for boundary cases

### Function Docstring

- [ ] The function's behavior is described
- The function's arguments are described:
  - [ ] Type
  - [ ] Purpose
  - [ ] Other assumptions (eg. if it's a number, what's the expected range?)
- The return value is described
  - [ ] Type
  - [ ] Other assumptions are documented
- The defensive assertions are documented using `Raises:`
  - [ ] Each assumption about an argument is checked with an assertion
  - [ ] Each assertion checks for _only one_ assumption about the argument
- [ ] Include 3 or more (passing!) doctests

### The Function

- [ ] The function's name describes it's behavior
- [ ] The function's name matches the file name
- [ ] The function has correct type annotations

### Do's

- [ ] Variable names help to understand the strategy
- [ ] Any comments are clear and describe the strategy
- [ ] Lines of code are spaced to help show different stages of the strategy

### Don'ts

- [ ] The function's strategy _is not_ described in the documentation
- [ ] Comments explain the _strategy_, **not** the _implementation_
- [ ] The function _does not_ have more comments than code
  - If it does, consider finding a new strategy or a simpler implementation

## Implementation

- [ ] The code passes the formatting checks
- [ ] The code passes all Ruff linting checks
- [ ] The code has no (reasonable) Pylint errors
  - In code review, you can decide when fixing a Pylint error is helpful and
    when it's too restricting.
- [ ] Variables are named with snake_case
- [ ] Variable names are clear and helpful
- [ ] The code follows the strategy as simply as possible
- [ ] The implementation is as simple as possible given the strategy
- [ ] The code includes defensive assertions
- [ ] Defensive assertions include as little logic as possible

### CODE REVIEW

File: solutions/tests/test_divide_two_numbers.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    def test_both_negative_numbers(self):
+        """It should correctly divide negative numbers"""
+        self.assertEqual(divide_numbers(-8, -2), 4)
+
+    def test_large_numbers(self):
+        """It should correctly divide large numbers"""
+        self.assertEqual(divide_numbers(2000000, 1000000), 2)
+
+    def test_zero_and_number(self):
+        """It should return zero if nominator is zero"""
```
Comment (suhrobmuboraksho) [inline solutions/tests/test_divide_two_numbers.py:33]: Typo: numerator*

**Review by suhrobmuboraksho — COMMENTED** [review]
Please address this issues. Thanks

**Review by suhrobmuboraksho — COMMENTED** [review]
All comments regarding changes were added to the solution. Good job @Djokwa

**Review by suhrobmuboraksho — APPROVED** [review]

File: solutions/divide_two_numbers.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
+"""
```
Comment (Alemayehu-Desta) [inline solutions/divide_two_numbers.py:1]: Use the Python interpreter and character encoding  at the top of the file

#!/usr/bin/env python3
# -*- coding: utf-8 -*-

File: solutions/tests/test_divide_two_numbers.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        with self.assertRaises(ValueError):
+            divide_numbers(46, 0)
+
+    def test_no_arguments(self):
+        """It should raise an error when no argument are provided"""
+        with self.assertRaises(TypeError):
+            divide_numbers()
+
+    def test_float_input(self):
+        """It should rise an error for float input"""
```
Comment (Alemayehu-Desta) [inline solutions/tests/test_divide_two_numbers.py:47]: typo error: replace 'rise' with 'raise'

File: solutions/tests/test_divide_two_numbers.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+    def test_large_numbers(self):
+        """It should correctly divide large numbers"""
+        self.assertEqual(divide_numbers(2000000, 1000000), 2)
+
+    def test_zero_and_number(self):
+        """It should return zero if numerator is zero"""
+        self.assertEqual(divide_numbers(0, 5), 0)
+
+    def test_zero_denominator_input(self):
```
Comment (Alemayehu-Desta) [inline solutions/tests/test_divide_two_numbers.py:36]: for consistency naming test methods, you can remove the input from test_zero_denominator_input. name it as tesst_zero_denominator.

**Review by Alemayehu-Desta — APPROVED** [review]
Left few minor comments

**Review by Manezhahm — DISMISSED** [review]

**Review by Khadijaramlawi — APPROVED** [review]

**Review by Manezhahm — APPROVED** [review]
